libpcsxcore/misc.c

```diff
@@ -169,6 +171,45 @@ static void fake_bios_gpu_setup(void)
 		GPU_writeData(gpu_data_def[i]);
 }
 
+/* Function for automatic patching according to GameID.
+ * It's possible that some of these games have no IDs, like some japanese games i encountered.
+ * I need to check whenever this matters or not for our games.
+ * (Plus it can still be activated in the menu)
+ * Let's hope the IDs are also not shared with other games ! (Homebrew, don't screw it up)
+ * */
+static void CheckforCDROMid_applyhacks()
+{
+	uint32_t i;
+
+	/* Apply hack battle fix for Inuyasha - Sengoku Otogi Kassen */
+	if (strncmp(CdromId, "SLPS03503", 9) == 0)
+	{
+		Config.VSyncWA = 1;
+		return;
+	}
+	
+	/* Apply Memory card hack for Codename Tenka. (The game needs one of the memory card slots to be empty) */
+	for(i=0;i<sizeof(MemorycardHack_db);i++)
+	{
+		if (strncmp(CdromId, MemorycardHack_db[i], 9) == 0)
+		{
+			/* Disable the second memory card slot for the game */
+			McdDisable[0] = 0;
+			McdDisable[1] = 1;
+		}
```

Comment:
What about `Config.Mcd2[0] = 0;` instead?

Reply:
That will work for psxbios.c but not for sio.c because McdDisable is used instead for the memory card functionality. Without it, the fix for Codename tenka will simply not work.